File: src/test/Bundle/Entity/Product.php
<?php

namespace test\Bundle\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
/**
 * Product
 *
 * @ORM\Table(name="product")
 * @ORM\Entity(repositoryClass="test\Bundle\Repository\ProductRepository")
 */
class Product
{
    /**
     * @var int
     *
     * @ORM\Column(name="id", type="integer")
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="AUTO")
     */
    private $idp;

    /**
     * @var float
     *@Assert\NotBlank()
     * @Assert\Regex(pattern="/^-?(?:\d+|\d*\.\d+)$/")
     * @ORM\Column(name="prix", type="float")
     */
    private $prix;

    /**
     * @var int
     *@Assert\NotBlank()
     * @ORM\Column(name="qte", type="integer")
     */
    private $qte;

    /**
     * @var string
     * @Assert\Length(min="4")
     * @ORM\Column(name="description", type="string", length=255)
     */
    private $description;



    /**
     * @var string
     *@Assert\Date
     *
     * @ORM\Column(name="date_expiration", type="string", length=255)
     */
    private $dateExpiration;

    /**
     * @var string

     *@Assert\NotBlank()
     * @Assert\Length(min="5")
     * @Assert\Regex(pattern="/^[a-zA-Z\s]+$/")
     * @ORM\Column(name="nom", type="string", length=255)
     */
    private $nom;

    /**
     * @return string
     */
    public function getNom()
    {
        return $this->nom;
    }

    /**
     * @param string $nom
     */
    public function setNom($nom)
    {
        $this->nom = $nom;
    }

    /**
     * @ORM\ManyToOne(targetEntity="Category")
     *  @Assert\NotBlank(message="Category Required")
     * @ORM\JoinColumn(name="cat_id",referencedColumnName="id")
     */
    private $category;

    /**
     * @var string
     *
     * @ORM\Column(name="image", type="string", length=255)
     */
    private $image;


    /**
     * @return string
     */
    public function getImage()
    {
        return $this->image;
    }

    /**
     * @param string $image
     */
    public function setImage($image)
    {
        $this->image = $image;
    }

    /**
     *
     * @return mixed
     */
    public function getCategory()
    {
        return $this->category;
    }

    /**
     * @param mixed $category
     */
    public function setCategory($category)
    {
        $this->category = $category;
    }

    /**
     * Get id
     *
     * @return int
     */
    public function getIdp()
    {
        return $this->idp;
    }

    /**
     * Set prix
     *
     * @param float $prix
     *
     * @return Product
     */
    public function setPrix($prix)
    {
        $this->prix = $prix;

        return $this;
    }

    /**
     * Get prix
     *
     * @return float
     */
    public function getPrix()
    {
        return $this->prix;
    }

    /**
     * Set qte
     *
     * @param integer $qte
     *
     * @return Product
     */
    public function setQte($qte)
    {
        $this->qte = $qte;

        return $this;
    }

    /**
     * Get qte
     *
     * @return int
     */
    public function getQte()
    {
        return $this->qte;
    }

    /**
     * Set description
     *
     * @param string $description
     *
     * @return Product
     */
    public function setDescription($description)
    {
        $this->description = $description;

        return $this;
    }

    /**
     * Get description
     *
     * @return string
     */
    public function getDescription()
    {
        return $this->description;
    }

    

    /**
     * Set dateExpiration
     *
     * @param string $dateExpiration
     *
     * @return Product
     */
    public function setDateExpiration($dateExpiration)
    {
        $this->dateExpiration = $dateExpiration;

        return $this;
    }

    /**
     * Get dateExpiration
     *
     * @return string
     */
    public function getDateExpiration()
    {
        return $this->dateExpiration;
    }
}

File: Integ4sur6/src/entite/Product.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package entite;
import java.sql.Date;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

/**
 *
 * @author Th3Doc
 */
@Entity
public class Product {
    @Id
    private int id;
    private String nom;
    private String image;
     private int category;
     private String cat;

    public String getCat() {
        return cat;
    }

    public void setCat(String cat) {
        this.cat = cat;
    }
    public Product(int id, String nom, String image, int category, float prix, int qte, String description, Date date_expiration) {
        this.id = id;
        this.nom = nom;
        this.image = image;
        this.category = category;
        this.prix = prix;
        this.qte = qte;
        this.description = description;
        this.date_expiration = date_expiration;
    }

    public int getCategory() {
        return category;
    }

    public void setCategory(int category) {
        this.category = category;
    }
    public Product(String nom, String image, float prix, int qte, String description, Date date_expiration) {
        this.nom = nom;
        this.image = image;
        this.prix = prix;
        this.qte = qte;
        this.description = description;
        this.date_expiration = date_expiration;
    }

    public Product(int id, String nom, String image, float prix, int qte, String description, Date date_expiration) {
        this.id = id;
        this.nom = nom;
        this.image = image;
        this.prix = prix;
        this.qte = qte;
        this.description = description;
        this.date_expiration = date_expiration;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }
    private float prix;
    private int qte;
    private String description;
    private java.sql.Date date_expiration; 

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public float getPrix() {
        return prix;
    }

    public void setPrix(float prix) {
        this.prix = prix;
    }

    @Override
    public String toString() {
        return "Product{" + "id=" + id + ", nom=" + nom + ", prix=" + prix + ", qte=" + qte + ", description=" + description + ", date_expiration=" + date_expiration + '}';
    }

    public int getQte() {
        return qte;
    }

    public void setQte(int qte) {
        this.qte = qte;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Date getDate_expiration() {
        return date_expiration;
    }

    public void setDate_expiration(Date date_expiration) {
        this.date_expiration = date_expiration;
    }
    public Product (){
        
    }
    

    public Product(int id, String nom, float prix, int qte, String description, Date date_expiration) {
        this.id = id;
        this.nom = nom;
        this.prix = prix;
        this.qte = qte;
        this.description = description;
        this.date_expiration = date_expiration;
    }

    public Product(String nom, float prix, int qte, String description, Date date_expiration) {
        this.nom = nom;
        this.prix = prix;
        this.qte = qte;
        this.description = description;
        this.date_expiration = date_expiration;
    }
    public Product( float prix, int qte, String description, Date date_expiration) {
       
        this.prix = prix;
        this.qte = qte;
        this.description = description;
        this.date_expiration = date_expiration;
    }
    
}
